platforms/ios/lib/WysiwygComposer/Sources/HTMLParser/Extensions/NSMutableAttributedString.swift

```diff
@@ -53,12 +53,18 @@ extension NSMutableAttributedString {
         }
     }
 
-    /// Removes any text that has been marked as discardable.
-    func removeDiscardableText() {
+    /// Finds any text that has been marked as discardable
+    /// and either replaces it with ZWSP if contained overlaps with text marked with a background style
+    /// or removes it otherwise
+    func replaceOrDeleteDiscardableText() {
         enumerateTypedAttribute(.discardableText) { (discardable: Bool, range: NSRange, _) in
             guard discardable == true else { return }
-
-            self.deleteCharacters(in: range)
+            let attributes = self.attributes(at: range.location, effectiveRange: nil)
```

Comment:
we can directly get the backgroundStyle alone here and avoid looking up the entire dictionary ourselves